iot/api/manager/src/main/java/com/google/cloud/iot/examples/DeviceRegistryExample.java

```diff
@@ -0,0 +1,338 @@
+ * cloud.google.com/iot. Since we are interacting with the device manager, we will only use the
+ * public keys. The private keys are used to sign JWTs to authenticate devices. See the MQTT client
+ * example in the developer guide for an example of this.
+ *
+ * <p>Finally, compile and run the example with:
+ *
+ * <pre>
+ * <code>
+ * $ mvn clean compile assembly:single
+ * $ mvn exec:java \
+ *       -Dexec.mainClass="com.google.cloud.iot.examples.DeviceRegistryExample" \
+ *       -Dexec.args="-project_id=my-project-id \
+ *                    -pubsub_topic=projects/my-project-id/topics/my-topic-id \
+ *                    -ec_public_key_file=/path/to/ec_public.pem \
+ *                    -rsa_certificate_file=/path/to/rsa_cert.pem"
+ * </code>
+ * </pre>
+ */
+public class DeviceRegistryExample {
+  // CloudIot service for administering Cloud IoT devices, registries and projects.
+  private CloudIot service;
+  // Path to the project and location: "projects/my-project-id/locations/us-central1"
+  private String projectPath;
+  // Path to the registry: "projects/my-project-id/location/us-central1/registries/my-registry-id"
+  private String registryPath;
+
+  /** Construct a new registry with the given name and pubsub topic inside the given project. */
+  public DeviceRegistryExample(
+      String projectId, String location, String registryName, String pubsubTopicPath)
+      throws IOException, GeneralSecurityException {
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(CloudIotScopes.all());
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    service = new CloudIot(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init);
+    projectPath = "projects/" + projectId + "/locations/" + location;
+    registryPath = projectPath + "/registries/" + registryName;
+
+    NotificationConfig notificationConfig = new NotificationConfig();
+    notificationConfig.setPubsubTopicName(pubsubTopicPath);
+
+    DeviceRegistry registry = new DeviceRegistry();
+    registry.setEventNotificationConfig(notificationConfig);
+    registry.setId(registryName);
+    service.projects().locations().registries().create(projectPath, registry).execute();
+  }
+
+  /** Delete this registry from Cloud IoT. */
+  public void delete() throws IOException {
+    System.out.println("Deleting: " + registryPath);
+    service.projects().locations().registries().delete(registryPath).execute();
+  }
+
+  /** Print all of the devices in this registry to standard out. */
+  public void listDevices() throws IOException {
+    List<Device> devices =
+        service
+            .projects()
+            .locations()
+            .registries()
+            .devices()
+            .list(registryPath)
+            .execute()
+            .getDevices();
```

Comment:
As with above, this is an Apiary pattern that I don't see an easy way around. I'll acknowledge and take a work item to see what can be done.